### ISSUE TITLE
running issues

### ISSUE THREAD

**Issue by wassfila** [TheSoundOfAIOSR/project_common#13]
* all repos not on latest : fixed

# Top repo
- [x] add tested csound version
- [x] add warning about long install time

# Production
- [x] readme path `rg_production\venv\Scripts\python rg_production\main.py`
- [x] missing matplotlib dependency
- [x] c-sound version missing , downloaded version in path `csound-windows_x86_64-6.16.2`

* errors screen freeze and `Error in setup`
```shell
  17: dac17 (Kopfh├Ârer (@System32\drivers\bthhfenum.sys,#2;%1 Hands-Free A)
PortAudio: selected output device 'Microsoft Sound Mapper - Output'
writing 2048 sample blks of 64-bit floats to dac0
SECTION 1:
Error in setup
Error in setup
Done loading stt
Setting up stt
Done loading sg
Error in setup
Error in setup
Error in setup
```

# Text to Sound
- [x] add missing dependency in prod requirements `asyncio`
- [x] fix command line call of rg_text_to_sound to be similar to other modules calls and remove `bash` and wsl dependency
- [x] fix module paths errors either with `sys.path.append` or other
- [x] reference the local path of `-r prod.txt` instead of `git+https://git@github.com/TheSoundOfAIOSR/rg_text_to_sound.git#'subdirectory=tts_pipeline`
see this section
https://github.com/TheSoundOfAIOSR/project_common/blob/9758ba352c3ea5004465de393207b22e87b22fc7/setup3.8.bat#L30

- bash command not compatible with windows script `source venv/bin/activate; cd tts_websocketserver/src; python -m tts_websocketserver.tts_server` as directory venv/bin does not exist 
- replaced with command `start ../../venv/Scripts/python -m tts_websocketserver/tts_server`
- run as `"../../venv/Scripts/python" -m tts_websocketserver/tts_server`
* `No module named tts_websocketserver/tts_server`
* run with `"venv/Scripts/python" tts_websocketserver\src\tts_websocketserver\tts_server.py`
* error : `ModuleNotFoundError: No module named 'tts_websocketserver.tts_pipeline'`
* activated venv 
* error : `ModuleNotFoundError: No module named 'tts_pipeline'`
* added `import sys
sys.path.append('tts_websocketserver')`
* error `ModuleNotFoundError: No module named 'tts_pipeline.pipelines'; 'tts_pipeline' is not a package`

**Comment by wassfila** [comment]
@atjessehill for info, I realised you're not in the RC's group so I invited you. The matoplotlib dependency is easy to add to the requirement.txt, I have it local, did not want to push on your repo maybe just let you do it.
The freeze and the error, I'm not sure if it's csound version related or if you've seen that already ?

**Comment by wassfila** [comment]
# issue when running with bash
```
D:\Dev\SoundHacking\project_common\rg_text_to_sound\tts_websocketserver\src>bash -c "source venv/bin/activate; cd tts_websocketserver/src; python -m tts_websocketserver.tts_server"
/bin/bash: venv/bin/activate: No such file or directory
/bin/bash: line 0: cd: tts_websocketserver/src: No such file or directory
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 163, in _run_module_as_main
    mod_name, _Error)
  File "/usr/lib/python2.7/runpy.py", line 119, in _get_module_details
    code = loader.get_code(mod_name)
  File "/usr/lib/python2.7/pkgutil.py", line 281, in get_code
    self.code = compile(source, self.filename, 'exec')
  File "/mnt/d/Dev/SoundHacking/project_common/rg_text_to_sound/tts_websocketserver/src/tts_websocketserver/tts_server.py", line 13
    async def process_text(self, text):
            ^
SyntaxError: invalid syntax
```

**Comment by wassfila** [comment]
after installing `asyncio` manually into the bash venv and `run.bat` again

```
D:\Dev\SoundHacking\project_common\rg_text_to_sound>bash -c "source venv/bin/activate; cd tts_websocketserver/src; python3 -m tts_websocketserver.tts_server"
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/mnt/d/Dev/SoundHacking/project_common/rg_text_to_sound/tts_websocketserver/src/tts_websocketserver/tts_server.py", line 1, in <module>
    from tts_websocketserver.tts_pipeline import get_pipeline
  File "/mnt/d/Dev/SoundHacking/project_common/rg_text_to_sound/tts_websocketserver/src/tts_websocketserver/tts_pipeline.py", line 1, in <module>
    from tts_pipeline.pipelines.waterfall.pipeline import WaterfallPipeline
ModuleNotFoundError: No module named 'tts_pipeline'
```

**Comment by wassfila** [comment]
After all the `bash` and wsl dependency could be the source of deviations between systems so removing it and having a symilar install run like other modules could be a clean up that solves this issue

**Comment by Leofltt** [comment]
@wassfila matplotlib is in the requirements, please pull most recent rg_production into project_common

**Comment by wassfila** [comment]
For info, more details about the tts fix , files to be edited
* https://github.com/TheSoundOfAIOSR/project_common/blob/main/setup3.8.bat
* https://github.com/TheSoundOfAIOSR/project_common/blob/main/setup.bat
* https://github.com/TheSoundOfAIOSR/project_common/blob/main/run.bat

![image](https://user-images.githubusercontent.com/1510763/128748872-04dc0b82-4acb-44bc-bd99-5954de213e7e.png)

**Comment by wassfila** [comment]
Taken from PR comments

- [ ] regression in speech to text as it was running before missing `pyaudio`

```
D:\Dev\SoundHacking\project_common\rg_speech_to_text>venv\Scripts\python TheSoundOfAIOSR\stt\interface\server_app.py       
Traceback (most recent call last):
  File "TheSoundOfAIOSR\stt\interface\server_app.py", line 9, in <module>
    from TheSoundOfAIOSR.stt.interface.wsserver import SimpleServerInterface
  File "D:\Dev\SoundHacking\project_common\rg_speech_to_text\.\TheSoundOfAIOSR\stt\interface\wsserver.py", line 4, in <module>
    from TheSoundOfAIOSR.stt.control.stt_sm import SpeechToTextSM
  File "D:\Dev\SoundHacking\project_common\rg_speech_to_text\.\TheSoundOfAIOSR\stt\control\stt_sm.py", line 6, in <module>    
    from TheSoundOfAIOSR.stt.control.speech_to_text import SpeechToText
  File "D:\Dev\SoundHacking\project_common\rg_speech_to_text\.\TheSoundOfAIOSR\stt\control\speech_to_text.py", line 7, in <module>
    from TheSoundOfAIOSR.audiointerface.capture import MicrophoneCaptureFailed
  File "D:\Dev\SoundHacking\project_common\rg_speech_to_text\.\TheSoundOfAIOSR\audiointerface\capture.py", line 3, in <module>
    import pyaudio
ModuleNotFoundError: No module named 'pyaudio'
```

**Comment by wassfila** [comment]
- [ ] windows anti-slash error

```
D:\Dev\SoundHacking\project_common>venv/Scripts/python -m tts_websocketserver/src/tts_websocketserver/tts_server
'venv' is not recognized as an internal or external command,
operable program or batch file.
```
![image](https://user-images.githubusercontent.com/1510763/129485146-fbd4bde0-b940-4f38-916d-9b41230b7f27.png)

**Comment by wassfila** [comment]
fixed with #26
